Services/Concurrency/RatedCounter.cs

```diff
@@ -177,19 +177,37 @@ public long GetPause()
             return false;
         }
 
+        /// <summary>
+        /// Get messages throughput.
+        /// </summary>
+        /// <returns>Throughput: messages per second</returns>
+        public double GetThroughputForMessages() => this.GetMessagesPerSecond(this.timestamps);
+
         private void LogThroughput()
         {
             if (this.log.LogLevel <= LogLevel.Debug && this.timestamps.Count > 1)
             {
-                double speed;
-                lock (this.timestamps)
+                double speed = this.GetMessagesPerSecond(this.timestamps);
+                this.log.Info(this.name, () => new { speed });
+            }
+        }
+
+        private double GetMessagesPerSecond(Queue<long> timestamps)
```

Comment:
remove this method, move all the logic to `GetThroughputForMessages`, and in `LogThroughput` call `GetThroughputForMessages()`

Reply:
for instance:
```
        /// <summary>
        /// Get messages throughput.
        /// </summary>
        /// <returns>Throughput: messages per second</returns>
        public double GetThroughputForMessages()
        {
            double speed = 0;
            lock (this.timestamps)
            {
                if (this.timestamps.Count > 1)
                {
                    // Time range in milliseconds
                    long time = this.timestamps.Last() - this.timestamps.First();

                    // Unit for speed is messages per second
                    speed = (1000 * (double) this.timestamps.Count / time * 10) / 10;
                }
            }

            return speed;
        }

        private void LogThroughput()
        {
            if (this.log.LogLevel <= LogLevel.Debug)
            {
                double speed = this.GetThroughputForMessages();
                this.log.Info(this.name, () => new { speed });
            }
        }
```